### DISCUSSION TITLE
How to delete a complete private tag

### DISCUSSION THREAD

**Discussion by bec1442** [fo-dicom/fo-dicom#1881] [Q&A] [answered]
I would like to delete the complete private sequence (3249,1020). The sequence contains several tags to include the creator (3249,0010).
Any sample code would be greatly appreciated.

**Comment by mrbean-bremen** [comment]
If I understand you correctly, you just want to delete a private sequence. I'm not sure what you mean by "The sequence contains several tags to include the creator (3249,0010)", but you can delete any sequence just by removing the sequence tag from the dataset:
```cs
var ds = DicomFile.Open("your_file.dcm").Dataset;
var seq = new DicomTag(0x3249, 0x20, "YOUR PRIVATE CREATOR");
ds.Remove(seq);
```
(error handling omitted)

**Reply by bec1442** [reply]
Many thanks mrbean
Apologize for not using correct terminology. The private seq. that I am trying to delete contains an item, each item has defined private tags. 
I hardcoded the creator, ensuring a new file is created and saved.
```
var dicomDataset = dicomFile.Dataset; 
var seq = new DicomTag(0x3249, 0x1020, "CSS Medical Systems 3249");
dicomDataset.Remove(seq);
```
no errors but seq not deleted.
The DCM file contains different creators to ensure the tag exist I did this:

 ```
DicomTag privateCreatorTag = new DicomTag(0x3249, 0x0010);
 if (dicomDataset.Contains(privateCreatorTag))
 {
     var privateCreator = dicomDataset.GetSingleValue<string>(privateCreatorTag);
     Console.WriteLine($"The creator is: {privateCreator}");
 }
 else
 {
        Console.WriteLine($"The creator does not exist");
 }
```
**output: "Creator does not exist"**
If I try this with different creator tag (e.i. DicomTag privateCreatorTag = new DicomTag(0x3253, 0x0010);
It works fine.
I am attaching a pix so may be can be understood better
![PrivateTag](https://github.com/user-attachments/assets/32c74ac9-92a2-47b0-a2e9-d2210d2407a9)

**Reply by mrbean-bremen** [reply, marked as answer]
Sorry, missed your reply. 
You didn't write (or I didn't understand) that the sequence you want to delete is inside another sequence.
In this case you have to navigate to that sequence first - the snippet above only works for top-level tags.
You would need something like:
```cs
var ds = DicomFile.Open("your_file.dcm").Dataset;
var fractionGroupSeqItems = ds.GetSequence(DicomTag.FractionGroupSequence).Items;
var refBeamSeqItems = fractionGroupSeqItems[0].GetSequence(DicomTag.ReferencedBeamSequence).Items;
var seqToRemove = new DicomTag(0x3249, 0x20, "CSS Medical Systems 3249");
refBeamSeqItems[0].Remove(seqToRemove);
```
(this is out of my head, so may not be completely correct)

**Reply by bec1442** [reply]
Awesome!
I wish I can buy you a coffee 😊
